### ISSUE TITLE
false positive: "bar" should not be a blacklisted name

### ISSUE THREAD

**Issue by lovasoa** [rust-lang/rust-clippy#4366]
Currently, clippy blacklists the name `bar`, so it can't be used as a variable name or a parameter name. However, *bar* is a common word in english, and can be used a descriptive variable name and not a placeholder.

An example from [the documentation of the crate `indicatif`](https://docs.rs/indicatif/0.11.0/indicatif/index.html#progress-bars-and-spinners):

```rust
use indicatif::ProgressBar;

let bar = ProgressBar::new(1000);
for _ in 0..1000 {
    bar.inc(1);
    // ...
}
bar.finish();
```

**Comment by flip1995** [comment]
In that case I would prefer `progress_bar` as the variable name.

The reason why `bar` is blacklisted is, that the names `foo`, `bar` and `baz` the main names for example or test code are. If I read `bar` in program code, I don't think of the english word `bar`, but the commonly used "to-lazy-to-think-of-a-name" name `bar`.

I see that some people may disagree. That's why the lint is [configurable](https://rust-lang.github.io/rust-clippy/master/index.html#blacklisted_name). 

Since having the common example names as a default SGTM and this can be configured, I'll go ahead and close this issue.
